src/attributes.jl

```diff
@@ -188,45 +193,82 @@ function h5readattr(filename, name::AbstractString)
 end
 
 
-struct Attributes
-    parent::Union{File,Object}
+struct AttributeDict <: AbstractDict{String,Any}
+    parent::Object
```

Comment:
Is `Union{File, Object}` redundant?

Reply:
Attributes attached to a file are really attached to the root group, so i made this explicit.